kotlinx-collections-immutable/src/main/kotlin/kotlinx/collections/immutable/implementations/immutableList/PersistentVectorBuilder.kt

```diff
@@ -26,12 +26,14 @@ private class Marker // TODO: Rename to MutabilityOwnership?
 class PersistentVectorBuilder<E>(private var vector: PersistentList<E>,
                                  private var vectorRoot: Array<Any?>?,
                                  private var vectorTail: Array<Any?>,
-                                 private var rootShift: Int) : AbstractMutableList<E>(), PersistentList.Builder<E> {
+                                 internal var rootShift: Int) : AbstractMutableList<E>(), PersistentList.Builder<E> {
     private var marker = Marker()
-    private var root = vectorRoot
-    private var tail = vectorTail
+    internal var root = vectorRoot
+    internal var tail = vectorTail
```

Comment:
If these are only read from outside makes sense to limit visibility of _setter_ to private

Reply:
+1